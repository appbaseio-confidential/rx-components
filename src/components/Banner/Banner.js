import React from 'react';

const Banner = () => <div>This is a banner</div>;

export default Banner;
