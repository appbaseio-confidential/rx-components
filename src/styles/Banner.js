import styled, { css } from 'react-emotion';

import { getThemeProperty } from '../utils';

const Banner = styled.a`
	border-radius: 2px;
	position: relative;
	display: inline-flex;
	overflow: hidden;
	box-shadow: ${({ theme }) => getThemeProperty(theme, 'colors', 'boxShadow')};
`;

const image = css`
	background-size: cover;
	background-position: center;
	width: 100%;
	height: 100%;
	position: relative;
`;

export default Banner;
export { image };
