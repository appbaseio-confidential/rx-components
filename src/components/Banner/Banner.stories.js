import React from 'react';
import { storiesOf } from '@storybook/react';
import Banner from './Banner';

const bannerImages = [
    'https://rukminim1.flixcart.com/flap/960/960/image/573190.jpg?q=50',
    'https://images-eu.ssl-images-amazon.com/images/G/31/img18/movies/1103933_750x600_gps_cn_qifq.jpg',
    'https://images-eu.ssl-images-amazon.com/images/G/31/img17/movies-tv/pagerefresh/5.jpg',
    'https://images-eu.ssl-images-amazon.com/images/G/31/img17/movies-tv/dec/1090291_750x600_Popular-English-TV-Shows.jpg',
];

storiesOf('Banner', module)
    .add('default', () => (
        <Banner
            href="#"
            style={{
                width: 550,
                height: 300,
            }}
            src={bannerImages[0]}
        />
    ))
    .add('in a row', () => (
        <div
            style={{
                display: 'grid',
                gridTemplateColumns: 'repeat(3, 1fr)',
                gridGap: 20,
            }}
        >
            <Banner
                href="#"
                style={{
                    height: 300,
                }}
                src={bannerImages[1]}
            />
            <Banner
                href="#"
                style={{
                    height: 300,
                }}
                src={bannerImages[2]}
            />
            <Banner
                href="#"
                style={{
                    height: 300,
                }}
                src={bannerImages[3]}
            />
        </div>
    ));
