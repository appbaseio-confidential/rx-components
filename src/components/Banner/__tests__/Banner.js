import React from 'react';
import renderer from 'react-test-renderer';
import Banner from '../Banner';

test('should match snapshot', () => {
	const tree = renderer
		.create(<Banner
				src="https://images-eu.ssl-images-amazon.com/images/G/31/img18/movies/1103933_750x600_gps_cn_qifq.jpg"
				href="movies/oscars"
				target="_blank"
				heightRatio={60}
				widthRatio={75}
		/>)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
