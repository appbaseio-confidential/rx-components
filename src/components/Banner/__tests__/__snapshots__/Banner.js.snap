// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-0 {
  background-size: cover;
  background-position: center;
  width: 100%;
  height: 0;
  padding-top: 66.6%;
}

.emotion-1 {
  border-radius: 2px;
  position: relative;
  overflow: hidden;
  box-shadow: 2px 2px 4px 0 rgba(0,0,0,.08);
}

<a
  className="emotion-1 emotion-2"
  href="movies/oscars"
  style={Object {}}
  target="_blank"
>
  <div
    className="emotion-0"
    style={
      Object {
        "backgroundImage": "url(https://images-eu.ssl-images-amazon.com/images/G/31/img18/movies/1103933_750x600_gps_cn_qifq.jpg)",
        "paddingTop": "80%",
      }
    }
  />
</a>
`;
